src/main/java/backtrace/io/BacktraceQueue.java

```diff
@@ -33,32 +40,46 @@ void addWithLock(BacktraceMessage message) {
     void unlock() {
         LOGGER.debug("Releasing semaphore..");
 
-        if(notEmptyQueue.getCount() == 0) {
-            notEmptyQueue.countUp();
+        if (processingLock.getCount() == 1) {
```

Comment:
Shouldn't we use `shouldHandleMessages` function instead?

Reply:
let's have it like this to separate this logic, maybe later we will change logic of `shouldHandleMessages ` and then it will affect here